File: leetcode/js/reverse-prefix-of-word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function (word, ch) {
  const idx = word.indexOf(ch);

  if (idx !== -1) {
    const reversed = [...word.slice(0, idx + 1)].reverse().join("");
    const rest = word.slice(idx + 1);
    word = reversed + rest;
  }

  return word;
};

File: leetcode/python/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        idx = word.find(ch)
        prefix = "".join(reversed(word[:idx+1]))

        return prefix + word[idx+1:]
    
class Solution1:
    def reversePrefix(self, word: str, ch: str) -> str:
        i = word.find(ch)
        if i == -1:
            return word
        return word[:i+1][::-1] + word[i+1:]